Docs: File transfer through the WebUI and Teleport Connect

mmcallister opened the pull request:
Addressing the request to add the WebUI options for copying files with the [Web UI  and Teleport Connect](https://github.com/gravitational/teleport/issues/39959).
(SCP via tsh scp is already present in the [protocols page](https://goteleport.com/docs/connect-your-client/introduction/#protocols).)

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

mmcallister committed:
scp through webui

zmb3 reviewed (commented):
I wonder if we should also mention moderated sessions in this guide.

Context: moderated sessions is a feature that allows you to require 1 or more moderators to be present and observing the session. When a `tsh` session requires moderation, file copying is completely disabled. If you are running a moderated session through the web UI instead of tsh you can copy files with moderator approval though.

ravicious commented on docs/pages/connect-your-client/web-ui.mdx:
```diff
@@ -31,3 +31,22 @@ You must have the `join_sessions` allow policy in a role you've been assigned to
+
+![drag and drop](../../img/upload-ui.png)
+
+### Teleport Connect
```
Thoughts on moving changes from this file to the introduction page you link to in the PR description? https://goteleport.com/docs/connect-your-client/introduction/#protocols

web-ui.mdx is largely for the Web UI, so it feels a bit awkward to mention Connect there.

I guess another problem is that we already have dedicated pages for Connect and the Web UI, so the question is whether these instructions should go to those specific pages individually or if they should be grouped together.

For the start though, I don't think there would be anything wrong with keeping them grouped, like we already do on the introduction page.

mmcallister replied:
Thanks! This may work better on the protocol page. Perhaps a new tabbed section for the **Transferring files (scp)** paragraph. The WebUI page is rather sparse in its current state, but that could be addressed separately.

ravicious commented on docs/img/download-with-connect.png:
Those two screenshots from Connect are quite heavy. 😄

I'd recommend two things:

1) Was it intentional to capture the extra background around the window? I'd drop it to reduce the size of the image as well, I don't think we typically include background in docs screenshots. Just a couple of weeks ago I discovered that in the built-in Screenshot app if you hold the Option key while taking the screenshot of a single window, macOS will not capture the shadow around the window.

2) [ImageOptim](https://imageoptim.com/mac) is a free app that I really recommend. It reduces image size without losing quality. Just by running this file through ImageOptim, I managed to drop its size by 31%.

mmcallister replied:
Thanks! I downloaded that and reduced the images considerably (~27%) and edited out the background. I grabbed them from the blog originally.

ravicious commented on docs/pages/connect-your-client/web-ui.mdx:
```diff
@@ -31,3 +31,22 @@ You must have the `join_sessions` allow policy in a role you've been assigned to
+
+### Teleport Connect
+
+Teleport Connect is a graphical user interface for Teleport that runs on Mac, Linux and Windows. The interface also allows you to upload local files to a remote server.
```
I'd also point out that the UI for that is accessible through the arrows in the top left of an ssh tab. Same for the Web UI. I remember when I was first using Teleport, I totally missed that upload/download is available because I totally missed those arrows!

mmcallister replied:
Oh nice! I didn't catch that. Thanks, I will add that as well

ravicious commented on docs/img/teleport-connect.png:
This file doesn't seem to be used anywhere in markdown.

mmcallister commented on docs/pages/connect-your-client/web-ui.mdx:
```diff
@@ -31,3 +31,22 @@ You must have the `join_sessions` allow policy in a role you've been assigned to
+
+![drag and drop](../../img/upload-ui.png)
+
+### Teleport Connect
```
✅ I'm moving this to the intro page under 'Transferring files'

mmcallister commented on docs/pages/connect-your-client/web-ui.mdx:
```diff
@@ -31,3 +31,22 @@ You must have the `join_sessions` allow policy in a role you've been assigned to
 For information about how to configure the `join_sessions` allow policy and participant modes for a role, see 
 [Configure an allow policy](../access-controls/guides/moderated-sessions.mdx#configure-an-allow-policy).
 
+## Secure Copy (SCP) files with the Web UI
```
✅

mmcallister commented on docs/pages/connect-your-client/web-ui.mdx:
```diff
@@ -31,3 +31,22 @@ You must have the `join_sessions` allow policy in a role you've been assigned to
+
+### Teleport Connect
+
+Teleport Connect is a graphical user interface for Teleport that runs on Mac, Linux and Windows. The interface also allows you to upload local files to a remote server.
```
✅

mmcallister commented on docs/img/teleport-connect.png:
✅ Changed to `teleport-connect-drag-drop.png`

mmcallister commented on docs/pages/connect-your-client/web-ui.mdx:
```diff
@@ -31,3 +31,22 @@ You must have the `join_sessions` allow policy in a role you've been assigned to
 For information about how to configure the `join_sessions` allow policy and participant modes for a role, see 
 [Configure an allow policy](../access-controls/guides/moderated-sessions.mdx#configure-an-allow-policy).
 
+## Secure Copy (SCP) files with the Web UI
```
✅ Added!

mmcallister committed:
file transfer options

mmcallister committed:
file transfer options

mmcallister committed:
file transfer options

mmcallister committed:
file transfer options

mmcallister commented:
Moved content to the introduction page, cleaned up and resized the images, and followed the tab formatting used in the rest of the page. https://docs-nr3f3fuu8-goteleport.vercel.app/docs/connect-your-client/introduction/#protocols

ravicious commented on docs/img/download-with-connect.png:
It looks like this screenshot got cut off too much and now the title bar at the top is not visible, compared to the other screenshot.

ravicious commented on docs/img/download-with-connect.png:
The size of both images can still be considerably reduced by running them through ImageOptim, either we have different ImageOptim settings (though IIRC I didn't change the default) or you ran ImageOptim on them before cutting out the background.

mmcallister replied:
Replaced and resized the photos

![resized-new-trsfr-imgs](https://github.com/gravitational/teleport/assets/6885322/95beebc6-6b21-4f4a-8fb5-8f6f5fd2ea60)

mmcallister replied:
I made more changes. I think I was overcomplicating it - basically, there are 2 ways in UIs to transfer files:

- Arrows in connect
- Arrows in webui

(Both support drag and drop for Upload files)

ravicious commented on docs/pages/connect-your-client/introduction.mdx:
```diff
@@ -250,6 +240,38 @@ cluster](../../img/use-teleport/kubernetes-login.png)
+</TabItem>
+<TabItem label="Web UI">
+
+You can also use the arrows in the top-left corner of the Web UI to upload and download files:
```
```suggestion
You can use the arrows in the top-left corner of the Web UI to upload and download files:
```

It'd be worth to mention that in the tab for Connect as well, since the UI for accessing file transfers is pretty much the same.

ravicious commented on docs/pages/connect-your-client/introduction.mdx:
```diff
@@ -250,6 +240,38 @@ cluster](../../img/use-teleport/kubernetes-login.png)
+
+You can upload and download files from systems using drag and drop:
+
+![Drag and drop in WebUI](../../img/teleport-connect-drag-drop.png)
```
```suggestion
![Drag and drop in Connect](../../img/teleport-connect-drag-drop.png)
```

ravicious reviewed (commented):
Submitted a couple of minor comments but it looks good overall.

mmcallister committed:
file transfer options

ravicious commented on docs/pages/connect-your-client/introduction.mdx:
```diff
@@ -250,6 +240,42 @@ cluster](../../img/use-teleport/kubernetes-login.png)
+</TabItem>
+<TabItem label="Teleport Connect">
+
+Teleport Connect allows you to transfer files to a remote server and runs on MacOS, Linux and Windows. You can use the arrows in the top-left corner of the Web UI to upload and download files:
```
```suggestion
Teleport Connect allows you to transfer files to a remote server and runs on MacOS, Linux and Windows. You can use the arrows in the top-left corner of an active SSH session to upload and download files:
```

mmcallister replied:
✅

ravicious commented on docs/img/download_files_connect.png:
I hate to be this pedantic but now those two Connect screenshots are the only screenshots with border shadow. 😭 It'd be nice to have them captured with no shadow, as I mentioned in my other comment: https://github.com/gravitational/teleport/pull/41798#discussion_r1608466254

But this is just a nit, I'm glad that they weight 10x less than they used to!

mmcallister replied:
Oh I was intentionally adding the scrollbar, in case someone had a bunch of files to transfer over. e.g.

![scrollbar-connect](https://github.com/gravitational/teleport/assets/6885322/9c79fc0b-29b8-4160-b282-cde71dc2d9e0)

But if you think it looks better without I can remove it. 

=)

ravicious replied:
AFAIK you can capture a screenshot with scrollbars but without border shadow, it shouldn't matter whether scrollbars are present or not.

mmcallister committed:
file transfer options - edit ct images

mmcallister committed:
file transfer options

mmcallister committed:
file transfer options - image name lint check

ptgott commented on docs/download-files-teleport-connect.png:
Let's put this in `docs/img` for consistency with other images

mmcallister replied:
Ah sorry, I had deleted this one from the doc and replaced it with `download_files_connect.png`. Removing it from the dir

ptgott commented on docs/pages/connect-your-client/introduction.mdx:
```diff
@@ -250,6 +240,42 @@ cluster](../../img/use-teleport/kubernetes-login.png)
+
+Teleport Connect allows you to transfer files to a remote server and runs on MacOS, Linux and Windows. You can use the arrows in the top-left corner of an active SSH session to upload and download files:
+
+![Download with connect](../../img/download_files_connect.png)
```
We'll need to update the image reference here

mmcallister replied:
I have been saving these in the `img` directory and they're all resolving for me:

```
-rw-r--r--@  1 mmcallister  staff    18024 May 22 12:20 upload_files_webui.png
-rw-r--r--@  1 mmcallister  staff    12826 May 22 12:21 download_files_webui.png
-rw-r--r--@  1 mmcallister  staff    50658 May 23 11:07 download_files_connect.png
-rw-r--r--@  1 mmcallister  staff    59520 May 23 11:53 upload_files_connect_drag_drop.png
/Users/mmcallister/Teleport/teleport/docs/img [copy_files_guide]
```

mmcallister committed:
file transfer options - delete unused image
